crates/viewer/re_component_fallbacks/src/component_fallbacks.rs

```diff
@@ -0,0 +1,458 @@
+use re_types::{Component as _, archetypes, components, datatypes};
+use re_viewer_context::{
+    ColormapWithRange, FallbackProviderRegistry, ImageDecodeCache, ImageInfo, ImageStatsCache,
+    QueryContext, TensorStats, TensorStatsCache, ViewerContext, auto_color_for_entity_path,
+};
+
+pub fn type_fallbacks(registry: &mut FallbackProviderRegistry) {
+    registry.register_type_fallback_provider::<components::Color>(|ctx| {
+        auto_color_for_entity_path(ctx.target_entity_path)
+    });
+    registry.register_type_fallback_provider::<components::Radius>(|_| 0.5.into());
+    registry.register_type_fallback_provider::<components::HalfSize2D>(|_| [0.5; 2].into());
+    registry.register_type_fallback_provider::<components::HalfSize3D>(|_| [0.5; 3].into());
+    registry.register_type_fallback_provider::<components::Vector2D>(|_| [1.0, 0.0].into());
+    registry.register_type_fallback_provider::<components::Vector3D>(|_| [1.0, 0.0, 0.0].into());
+    registry.register_type_fallback_provider::<components::Timestamp>(|_| {
+        components::Timestamp(0.into())
+    });
+    registry.register_type_fallback_provider::<components::SeriesVisible>(|_| true.into());
+    registry.register_type_fallback_provider::<components::ViewCoordinates>(|_| {
+        archetypes::Pinhole::DEFAULT_CAMERA_XYZ
+    });
+    registry.register_type_fallback_provider::<components::Resolution>(|ctx| {
+        // If the Pinhole has no resolution, use the resolution for the image logged at the same path.
+        // See https://github.com/rerun-io/rerun/issues/3852
+        resolution_of_image_at(ctx.viewer_ctx(), ctx.query, ctx.target_entity_path)
+            // Zero will be seen as invalid resolution by the visualizer, making it opt out of visualization.
+            // TODO(andreas): We should display a warning about this somewhere.
+            // Since it's not a required component, logging a warning about this might be too noisy.
+            .unwrap_or(components::Resolution::from([0.0, 0.0]))
+    });
+}
+
+pub fn archetype_fallbacks(registry: &mut FallbackProviderRegistry) {
```

Comment:
they're not really fallbacks for archetypes but fallbacks for archetype fields if we want to contrast them with type fallbacks. So probably `field_fallbacks` or `archetype_field_fallbacks`
Counter point is that we call those fields nowadays often just "components", but by that reasoning we have `component_type_fallbacks` and `component_fallbacks` which is more confusing not less :thinking: